### ISSUE TITLE
Disable nested markdown via configuration

### ISSUE THREAD

**Issue by andrewplummer** [mdx-js/mdx#1946]
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

In the playground editor input the following:

```
<p>
  Hello world
</p>
<div>
  <span>
    one
  </span>
  <span>
    two
  </span>
</div>
```

### Solution

Expected compiled result:

```
<p>
  Hello world
</p>
<div>
  <span>
    one
  </span>
  <span>
    two
  </span>
</div>
```

### Alternatives

---

I am forced to comment here now that you have shut down discussion in #1798 (with apologies to @ksinas for getting his thread locked).

I am sincerely confused by your reaction to what was a relevant and legitimate discussion. You accused me of simply wanting to argue then proceeded to do exactly that while ignoring my attempts at being constructive. I was a bit sharp up front and I apologize for that, but I wasn't expecting the reaction I got.

> You seem to have missed or be unclear that the original ask, skipping some paragraphs, is supported/configurable today with plugins

No I'm very clear on that point. I was asking follow up questions to try to understand the thinking behind this as well as a clarification. The current behavior (content inside another tag resulting in nested `<p>` tags and invalid markup unless on a single line or wrapped in `{}` braces) truly **is** counterintuitive and I legitimately thought I was doing it wrong  at first.

> You also seem to unclear what MDX is, it is JSX mixed with Markdown. You seem to want something that isn't JSX mixed with Markdown, if so you may want another language.

Let's not be snarky, you don't *actually* think that I am unclear what MDX is. And I will not pretend that you don't understand that "JSX mixed with Markdown" can be done in different ways and does not mean just one thing. What you seem to be saying instead is "your use case is invalid and we won't support that". And that's fine! But as a suggestion for the future: something like "from our experience more users want markdown nested inside JSX" or even just "this is our vision for MDX, sorry" would be more constructive than claiming I don't know what MDX is (especially when I am a big fan!)

To be clear, I am opening this request 100% in sincerity, not to argue or troll. I don't think it's an unfair or unreasonable request that something as fundamental as the above use case be a core configuration, not relegated to a plugin (which by the way doesn't exist, your linked answer amounted to basically "go write your own").

In anticipation to this request being simply closed and locked and this feature never implemented (which seems likely) I will say that for my part I will probably be looking for a migration path back to v1. Your attitude hasn't done much to inspire my confidence but of course that's not a valid reason to choose whether to use open source software. At the end of the day MDX v2 simply **doesn't do** what I want and MDX v1 did perfectly. Migrating back is just simpler than writing my own custom plugin.

I got the sense from your website that you guys are very proud of v2 (which was just released) and I want to say that you should be! You've spent countless hours on this project and MDX is overall fantastic. I will absolutely be using it in some form going forward (regardless if it's v1 or v2). I understand that it can be very frustrating dealing with users that sound entitled when they have issues with the software you not only spend a lot of time on but also provide for free! If I came across that way then I really do apologize. 

Please feel free now to close and lock this issue, probably as argumentative or just opinion (aren't all feature requests opinion?). At any rate I won't post about it again.

Thank you!

**Comment by wooorm** [comment]
> I am opening this request 100% in sincerity, not to argue or troll

This request has been opened, discussed, and closed, in #1798. The behavior around how markdown and JSX interleave has been discussed for about 2 years leading up to version 2.
You disagree with the outcome which is fine. A couple people disagreeing isn’t going to change this though.

> your linked answer amounted to basically "go write your own"

It is expected that people who have custom needs for MDX-like content use plugins.

> If I came across that way then I really do apologize.

Thanks <3

**Comment by wooorm** [comment]
Duplicate of #1798
